modules/core/src/main/java/org/apache/ignite/internal/processors/query/QueryUtils.java

```diff
@@ -1618,6 +1619,16 @@ public static boolean removeField(QueryEntity entity, String alias) {
         return entity.getFields().remove(fieldNameByAlias(entity, alias)) != null;
     }
 
+    /**
+     * @param qry Query.
+     * @param timeout Timeout.
+     * @param timeUnit Time units.
+     */
+    public static void copyQueryTimeout(SqlFieldsQuery qry, int timeout, TimeUnit timeUnit) {
+        if (timeout >= 0)
+            qry.setTimeout(timeout, timeUnit);
+    }
```

Comment:
Maybe it would be better to provide method like this:
```suggestion
    public static SqlFieldsQuery withQueryTimeout(SqlFieldsQuery qry, int timeout, TimeUnit timeUnit) {
        if (timeout >= 0)
            qry.setTimeout(timeout, timeUnit);
            
        return qry;
    }
```

And change all usages to method-chaining-way where possible (e.g.  `withQueryTimeout(qry, timout, timeUnit).setOtherImportantProperty()`)